debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/ModularityHelper.java

```diff
@@ -340,4 +342,18 @@ public static int iterateClassLoaderClasses(PrintStream out, ClassIteratorFilter
 		return count;
 	}
 
+	public static String getModuleName(J9ModulePointer modulePtr) throws CorruptDataException {
+	    String moduleName = null;
+	    if (modulePtr != null) {
+	        Object mn = modulePtr.moduleName();
+	        if (mn instanceof J9ObjectPointer) {
+	            moduleName = J9ObjectHelper.stringValue((J9ObjectPointer)mn);
+	        } else if (mn instanceof J9UTF8Pointer) {
+	            moduleName = J9UTF8Helper.stringValue((J9UTF8Pointer)mn);
+	        } else {
+	            throw new CorruptDataException("Invalid module name data type encountered: " + mn.toString());
+	        }
+	    }
+	    return moduleName;
```

Comment:
Tabs will need to be used here. We should perform a brief sanity check using `jdmpview + core file` to confirm that this helper is functioning correctly.

Reply:
There is something wrong with my auto indent functionality between vim/VSCode; really need to nail that down. Thanks for catching that. I'll test it too.

Reply:
This function won't work on older core files because `J9ModulePointer::moduleName()` is generated for the build of `jdmpview` based on they `moduleName` type in the JDK build. i.e. the signature is (and from now on will be):
```
@com.ibm.j9ddr.GeneratedFieldAccessor(offsetFieldName="_moduleNameOffset_", declaredType="J9UTF8*")
public J9UTF8Pointer moduleName() throws CorruptDataException {...}
```

Using an older core file doesn't change the fact that this function returns a `J9UTF8Pointer`.

Reply:
On the contrary, the blob in an old core records the fact that `moduleName` was `J9Object*`.
```
$ omr_blob_reader jdk-21/lib/default/j9ddr.dat
...
Struct name: J9Module
 no superName
 sizeOf: 64
 fieldCount: 9
 constCount: 0
 Field declaredName: moduleName
  declaredType: struct J9Object*
  offset: 0
```

Also, the indentation is a mix of tabs and spaces; here, the indentation should be exclusively tabs.

Reply:
I'm not sure there is precedent for changing the type of a field used in DDR code. We may need to use a new name for the J9UTF8 flavour.

Reply:
> On the contrary, the blob in an old core records the fact that moduleName was J9Object*

I understand that this is the case. My point was that using a newer `jdmpview` with the changed type to examine an older core, results in the `J9ModulePointer::moduleName()` to cause a:

```
java.lang.NoSuchMethodError: com/ibm/j9ddr/vm29/pointer/generated/J9ModulePointer.moduleName()Lcom/ibm/j9ddr/vm29/pointer/generated/J9UTF8Pointer; (loaded from <Unknown> by J9DDRClassloader for com.ibm.j9ddr.vm29.) called from class com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper (loaded from <Unknown> by J9DDRClassloader for com.ibm.j9ddr.vm29.).
```
because of this difference.

Reply:
Yes, the signature of the reference is baked into `ModularityHelper.class`.